lib/openstack/amqp/openstack_qpid_event_monitor.rb

```diff
@@ -13,10 +13,25 @@ def self.available?(options = {})
     OpenstackQpidConnection.available? && test_connection(options)
   end
 
+  def self.connection_parameters(options = {})
+    username = options[:username] if options.key? :username
+    password = options[:password] if options.key? :password
+    [
+      options[:hostname],
+      options[:port],
+      username,
+      password
+    ].compact
```

Comment:
Are you sure you want compact?  won't that possibly break the parameter order if something like the port is missing?